R/agglomerate.R

```diff
@@ -399,10 +399,9 @@ setMethod(
     if( remove_empty_ranks ){
         # Get rowData
         rd <- rowData(x)
-        # Does teh column include data?
-        columns_including_data <- apply(rd, 2, function(x){!all(is.na(x))})
-        # Subset data so that it includes only columns that include data
-        rd <- rd[, columns_including_data]
+        rank.cols <- taxonomyRanks(x)
+        # Subset data so that it includes only rank columns that include data
+        rd <- rd[, !apply(rd[, rank.cols, drop = FALSE], 2, function(x) all(is.na(x)))]
         # Assign it back to SE
         rowData(x) <- rd
```

Comment:
This should be somethig like this probably

```
# Get columns that include taxonomy infromation
rank_cols <- taxonomyRanks(x)
# Get rowData with only taxonomy
rd <- rowData(x)[ , rank_cols, drop = FALSE]
# Remove taxonomy from rowData
rowData(x) <- rowData(x)[ , !colnames(x) %in% rank_cols, drop = FALSE]
# Subset data so that it includes only rank columns that include data
empty_ranks <-!apply(rd, 2, function(x) all(is.na(x)))
rd <- rd[ , empty_ranks, drop = FALSE]
# Assing taxonomy back to SE
rowData(x) <- cbind(rowData(x), rd)
```

Reply:
Thanks.